unit.js

```diff
@@ -0,0 +1,82 @@
+function Unit(name) {
+    const _MAX_HEALTH = 100;
+    const _XP_PER_LEVEL = 1000;
+
+    let _health = _MAX_HEALTH;
+    let _name = name;
+    let _level = 1;
+    let _xp = 0;
+
+    this.resistance = 0;
+    this.pointsXP = {
+        default: 250,
+        damage: 500
+    };
+
+    this.getName = function() {
+        return _name;
+    };
+
+    this.getLevel = function() {
+        return _level;
+    };
+
+    this.getXP = function() {
+        return _xp;
+    };
+
+    this.getHealth = function() {
+        return _health;
+    };
+
+    this.isAlive = function() {
+        return _health > 0;
+    };
+
+    this.addHealth = function(value) {
+        _health += value;
+        if (_health > _MAX_HEALTH) {
+            _health = _MAX_HEALTH;
+        }
+    };
+
+    this.takeDamage = function(value) {
+        let damagePoints = value * (1 - this.resistance);
+        damagePoints = Math.round(damagePoints * 100) / 100;
+        if (_health > damagePoints) {
+            _health = _health - damagePoints;
+        } else {
+            _health = 0;
+        }
+
+        // _earnExperience(this.pointsXP.damage);   // I'm not a gamer... but why should it earn xp for taking damage?
```

Comment:
Traumas and hurtful experience is a lessons, and some people saying `learn from your mistakes` ;)